refactor(portal): Add website_link component

jamilbk opened the pull request:
https://github.com/firezone/firezone/pull/3579#discussion_r1496082754

jamilbk committed:
Add website_link component

jamilbk committed:
Make link_style() available everywhere

bmanifold commented on elixir/apps/web/lib/web/live/relay_groups/new_token.ex:
```diff
@@ -197,10 +197,10 @@ defmodule Web.RelayGroups.NewToken do
-                href="https://www.firezone.dev/kb/administer/troubleshooting#relay-not-connecting"
-              >relay troubleshooting guide</.link>.
+              Relay not connecting? See our
+              <.website_link href="/kb">
```
I took a quick look and saw that we don't have a section for Relay troubleshooting in our current docs, but I was wondering if we wanted to go ahead and have this link go to: `/kb/administer/troubleshooting` ?

jamilbk replied:
Yeah, it's a bit weird keeping these up to date but I fixed it

bmanifold reviewed (approved):
LGTM.  One small question, but it's not blocking

jamilbk committed:
Move sales contact to lead intake form

jamilbk committed:
Relay troubleshooting link:

jamilbk commented on elixir/apps/web/lib/web/live/settings/billing.ex:
```diff
@@ -72,6 +63,9 @@ defmodule Web.Settings.Billing do
             <:label>Current Plan</:label>
             <:value>
               <%= @account.metadata.stripe.product_name %>
+              <%= if @account.metadata.strine.product_name != "Enterprise" do %>
```
@AndrewDryga It would be preferred if we can send them to [contact sales](https://www.firezone.dev/contact/sales) because then they'll enter our Sales pipeline

AndrewDryga replied:
But that form is for new customers, we need to somehow pass their account ID. And they should go through a slightly different pipeline as existing customers

jamilbk replied:
When they fill that out, Hubspot automatically links us to all of their activity across our docs and any other contact we've had with them. From there I can more easily look them up.

I think the fix here is to just turn on Hubspot's Service desk feature to route emails from `support@`. Currently they get routed by Google which doesn't show us their email address for replying to.

jamilbk replied:
Email routing is now fixed. So I'll revert this change

jamilbk committed:
Add utm_source to email links

jamilbk commented on elixir/apps/web/lib/web/components/core_components.ex:
```diff
@@ -1214,10 +1214,19 @@ defmodule Web.CoreComponents do
 
       ---
       Please do not remove this part of the email.
+      Account Name: #{account.name}
```
These will help me quickly match them in Hubspot and Slack

AndrewDryga commented on elixir/apps/web/lib/web/components/navigation_components.ex:
```diff
@@ -308,4 +308,32 @@ defmodule Web.NavigationComponents do
+  slot :inner_block, required: true
+  attr :rest, :global
+
+  # XXX: We could add automated testing to make sure these don't return 404s, but that will fail
```
Please use TODO: or better create a GitHub issue for this, because it won't be trivial to implement this and nobody greps codebase by XXX right now

jamilbk replied:
Updated to remove. I don't think this will be a major issue and probably not worth investing the time to implement and maintain it

jamilbk committed:
Merge branch 'main' into feat/docs-link

conectado committed:
fix(connlib): set the real packet length before putting it into the device (#3718)

This was fixed at some point in the feature branch but was lost to time.

This is preventing macos from working(and might be causing some issues
in other platforms)

ReactorScram committed:
refactor: move BUNDLE_ID since it won't be Windows-specific (#3706)

It is still client-specific, but this was the closest place I could find
in connlib to put it.
A hypothetical GUI / .deb / systemd-involved gateway would need to be
"dev.firezone.gateway"

jamilbk committed:
Address feedback

jamilbk committed:
Merge remote-tracking branch 'origin/main' into feat/docs-link
